diff -Nru orig/arcgisgeocode/src/Makevars.win.in new/arcgisgeocode/src/Makevars.win.in
--- orig/arcgisgeocode/src/Makevars.win.in	2026-08-01 22:35:31
+++ new/arcgisgeocode/src/Makevars.win.in	2026-08-01 22:37:06
@@ -1,4 +1,4 @@
-TARGET = $(subst 64,x86_64,$(subst 32,i686,$(WIN)))-pc-windows-gnu
+TARGET = @WINDOWS_TARGET@
 
 TARGET_DIR = ./rust/target
 LIBDIR = $(TARGET_DIR)/$(TARGET)/@LIBDIR@
@@ -19,7 +19,7 @@
 	touch $(TARGET_DIR)/libgcc_mock/libgcc_eh.a
 
 	if [ "$(NOT_CRAN)" != "true" ]; then \
-		if [ -f ./rust/vendor.tar.xz ]; then \
+		if [ -f ./rust/novendor.tar.xz ]; then \
 			tar xf rust/vendor.tar.xz && \
 			mkdir -p $(CARGOTMP) && \
 			cp rust/vendor-config.toml $(CARGOTMP)/config.toml; \
diff -Nru orig/arcgisgeocode/src/rust/Cargo.lock new/arcgisgeocode/src/rust/Cargo.lock
--- orig/arcgisgeocode/src/rust/Cargo.lock	2026-08-01 22:35:31
+++ new/arcgisgeocode/src/rust/Cargo.lock	2026-08-01 22:37:34
@@ -48,12 +48,6 @@
 checksum = "5c8214115b7bf84099f1309324e63141d4c5d7cc26862f97a0a857dbefe165bd"
 
 [[package]]
-name = "build-print"
-version = "0.1.1"
-source = "registry+https://github.com/rust-lang/crates.io-index"
-checksum = "4a2128d00b7061b82b72844a351e80acd29e05afc60e9261e2ac90dca9ecc2ac"
-
-[[package]]
 name = "bumpalo"
 version = "3.17.0"
 source = "registry+https://github.com/rust-lang/crates.io-index"
@@ -225,9 +219,9 @@
 
 [[package]]
 name = "extendr-api"
-version = "0.8.0"
+version = "0.8.2"
 source = "registry+https://github.com/rust-lang/crates.io-index"
-checksum = "cae12193370e4f00f4a54b64f40dabc753ad755f15229c367e4b5851ed206954"
+checksum = "e002a15028421248374d3beb66d1802ce70d738b666a043a1f4dfffad14bc595"
 dependencies = [
  "extendr-ffi",
  "extendr-macros",
@@ -238,18 +232,15 @@
 
 [[package]]
 name = "extendr-ffi"
-version = "0.8.0"
+version = "0.8.2"
 source = "registry+https://github.com/rust-lang/crates.io-index"
-checksum = "b48f96b7a4a2ff009ad9087f22a6de2312731a4096b520e3eb1c483df476ae95"
-dependencies = [
- "build-print",
-]
+checksum = "acf3ace33b895bdfc13d02fd7486b26dc71078c256f3399f6272f8bd70fc3a1e"
 
 [[package]]
 name = "extendr-macros"
-version = "0.8.0"
+version = "0.8.2"
 source = "registry+https://github.com/rust-lang/crates.io-index"
-checksum = "fdbbac9afddafddb4dabd10aefa8082e3f057ec5bfa519c7b44af114e7ebf1a5"
+checksum = "f6b27baf9abffe403f3051efea3c19a0f315be9ec5f1c80efbade1aee5ccccbc"
 dependencies = [
  "proc-macro2",
  "quote",
diff -Nru orig/arcgisgeocode/src/rust/Cargo.toml new/arcgisgeocode/src/rust/Cargo.toml
--- orig/arcgisgeocode/src/rust/Cargo.toml	2026-08-01 22:35:31
+++ new/arcgisgeocode/src/rust/Cargo.toml	2026-08-01 22:37:17
@@ -10,7 +10,7 @@
 name = 'arcgisgeocode'
 
 [dependencies]
-extendr-api = { version = "0.8.0", features = ["serde"] }
+extendr-api = { version = "0.8.2", features = ["serde"] }
 rust_iso3166 = "0.1.12"
 serde = "*"
 serde_esri = { git = "https://github.com/josiahparry/serde_esri" }
diff -Nru orig/arcgisgeocode/tools/config.R new/arcgisgeocode/tools/config.R
--- orig/arcgisgeocode/tools/config.R	2026-08-01 22:35:31
+++ new/arcgisgeocode/tools/config.R	2026-08-01 22:37:06
@@ -6,7 +6,7 @@
 env_not_cran <- Sys.getenv("NOT_CRAN")
 
 # check if the vendored zip file exists
-vendor_exists <- file.exists("src/rust/vendor.tar.xz")
+vendor_exists <- FALSE
 
 is_not_cran <- env_not_cran != ""
 is_debug <- env_debug != ""
@@ -63,8 +63,19 @@
 # read as a single string
 mv_txt <- readLines(mv_fp)
 
+.windows_target <- if (grepl("aarch", R.version$platform)) {
+  "aarch64-pc-windows-gnullvm"
+} else if (grepl("clang", Sys.getenv("R_COMPILED_BY"))) {
+  "x86_64-pc-windows-gnullvm"
+} else if (grepl("i386", R.version$platform)) {
+  "i686-pc-windows-gnu"
+} else {
+  "x86_64-pc-windows-gnu"
+}
+
 # replace placeholder values
 new_txt <- gsub("@CRAN_FLAGS@", .cran_flags, mv_txt) |>
+  gsub("@WINDOWS_TARGET@", .windows_target, x = _) |>
   gsub("@PROFILE@", .profile, x = _) |>
   gsub("@CLEAN_TARGET@", .clean_targets, x = _) |>
   gsub("@LIBDIR@", .libdir, x = _)
