diff -Nru orig/arrow/configure.win new/arrow/configure.win
--- orig/arrow/configure.win	2026-07-27 20:58:26
+++ new/arrow/configure.win	2026-07-27 20:58:57
@@ -48,8 +48,9 @@
 
 
 function configure_binaries() {
-  # Try to find/download a C++ Arrow binary,
-  "${R_HOME}/bin${R_ARCH_BIN}/Rscript.exe" "tools/nixlibs.R" $VERSION
+  # Download a static build of libarrow and dependencies from
+  # https://github.com/r-windows/bundles
+  "${R_HOME}/bin${R_ARCH_BIN}/Rscript.exe" "tools/winlibs.R"
   # If binary not found, script exits nonzero
   if [ $? -ne 0 ]; then
     _LIBARROW_FOUND="false"
@@ -69,15 +70,7 @@
             -lz -lws2_32 -lnghttp2 -ldbghelp"
 
   # Set the right flags to point to and enable arrow/parquet
-  if [ -d "windows/r-libarrow-windows-x86_64-$VERSION" ]; then
-    RWINLIB="../windows/r-libarrow-windows-x86_64-$VERSION"
-  else
-    # It's possible that the version of the libarrow binary is not identical to the
-    # R version, e.g. if the R build is a patch release, so find what the dir is
-    # actually called. If there is more than one version present, use the one
-    # with the highest version:
-    RWINLIB="../windows/$(ls windows/ | grep ^r-libarrow- | tail -n 1)"
-  fi
+  RWINLIB="../windows/arrow"
 
   # NOTE: If you make changes to the libraries below, you should also change
   # ci/scripts/r_windows_build.sh and ci/scripts/PKGBUILD
@@ -85,8 +78,7 @@
               -DARROW_DS_STATIC -DARROW_ACERO_STATIC -DARROW_R_WITH_PARQUET \
               -DARROW_R_WITH_COMPUTE -DARROW_R_WITH_ACERO \
               -DARROW_R_WITH_DATASET -DARROW_R_WITH_JSON"
-  PKG_LIBS="-L${RWINLIB}/lib"'$(subst gcc,,$(COMPILED_BY))$(R_ARCH) '
-  PKG_LIBS="$PKG_LIBS -L${RWINLIB}/lib"'$(R_ARCH)$(CRT) '
+  PKG_LIBS="-L${RWINLIB}/lib"
   PKG_LIBS="$PKG_LIBS -larrow_dataset -larrow_acero -lparquet -larrow_compute -larrow -larrow_bundled_dependencies \
             -lutf8proc -lsnappy -lz -lzstd -llz4 -lbz2 ${BROTLI_LIBS} -lole32 \
             ${MIMALLOC_LIBS} ${OPENSSL_LIBS}"
diff -Nru orig/arrow/tools/winlibs.R new/arrow/tools/winlibs.R
--- orig/arrow/tools/winlibs.R	1970-01-01 01:00:00
+++ new/arrow/tools/winlibs.R	2026-07-27 20:58:57
@@ -0,0 +1,24 @@
+# Download static build of libarrow + dependencies from:
+# https://github.com/r-windows/bundles/releases
+if (!file.exists("windows/arrow/include/arrow/api.h")) {
+  bundle <- "25.0.0"
+  if (grepl("aarch64", R.version$platform)) {
+    toolchain <- "clang-aarch64"
+  } else if (grepl("clang", Sys.getenv("R_COMPILED_BY"))) {
+    toolchain <- "clang-x86_64"
+  } else if (getRversion() >= "4.2") {
+    toolchain <- "ucrt-x86_64"
+  } else {
+    stop("This package requires R >= 4.2 (rtools42 or newer)")
+  }
+  unlink("windows", recursive = TRUE)
+  url <- sprintf(
+    "https://github.com/r-windows/bundles/releases/download/arrow-%s/arrow-%s-%s.tar.xz",
+    bundle, bundle, toolchain
+  )
+  download.file(url, basename(url), quiet = TRUE)
+  dir.create("windows", showWarnings = FALSE)
+  untar(basename(url), exdir = "windows", tar = "internal")
+  unlink(basename(url))
+  file.rename(sprintf("windows/arrow-%s-%s", bundle, toolchain), "windows/arrow")
+}
