--- a/mzR/src/boost_aux/libs/nowide/src/iostream.cpp
+++ b/mzR/src/boost_aux/libs/nowide/src/iostream.cpp
@@ -222,7 +222,7 @@ namespace detail {
     {
     }
 
-    winconsole_istream::winconsole_istream(winconsole_ostream* tieStream=0) : std::istream(0)
+    winconsole_istream::winconsole_istream(winconsole_ostream* tieStream) : std::istream(0)
     {
         HANDLE h = GetStdHandle(STD_INPUT_HANDLE);
         d.reset(new console_input_buffer(h));
diff --git a/mzR/src/pwiz/utility/misc/Filesystem.cpp b/mzR/src/pwiz/utility/misc/Filesystem.cpp
index 5f1a7c473..f5289d90c 100644
--- a/mzR/src/pwiz/utility/misc/Filesystem.cpp
+++ b/mzR/src/pwiz/utility/misc/Filesystem.cpp
@@ -138,7 +138,7 @@ extern "C"
         );
 
     PVOID GetLibraryProcAddress(PSTR LibraryName, PSTR ProcName) {
-        return GetProcAddress(GetModuleHandleA(LibraryName), ProcName);
+        return (PVOID)GetProcAddress(GetModuleHandleA(LibraryName), ProcName);
     }
 
     typedef struct __PUBLIC_OBJECT_TYPE_INFORMATION {
