diff --git a/salso/tools/configure.R b/salso/tools/configure.R
index 9d6fc40..26aef02 100755
--- a/salso/tools/configure.R
+++ b/salso/tools/configure.R
@@ -397,7 +397,7 @@ render_platform_makevars <- function(
       note(
         "Non-GNU Windows host detected; cross-compiling with --target=x86_64-pc-windows-gnu."
       )
-      target_decl <- "TARGET = x86_64-pc-windows-gnu"
+      target_decl <- paste("TARGET =", windows_target())
       rust_target_lib <- "$(RUST_DIR)/target/$(TARGET)/release/librust.a"
       target_flag <- " --target=$(TARGET)"
       cargo_pre <- ""
@@ -430,6 +430,18 @@ render_platform_makevars <- function(
   }
 }
 
+windows_target <- function(){
+  if(grepl("aarch", R.version$platform)){
+    "aarch64-pc-windows-gnullvm"
+  } else if(grepl("clang", Sys.getenv('R_COMPILED_BY'))){
+    "x86_64-pc-windows-gnullvm"
+  } else if(grepl("i386", R.version$platform)){
+    "i686-pc-windows-gnu"
+  } else {
+    "x86_64-pc-windows-gnu"
+  }
+}
+
 note("*** starting configure.R ***")
 note(sprintf("Timestamp: %s", Sys.time()))
 
